### ISSUE TITLE
Attempt to read from field 'com.mbientlab.metawear.impl.ResponseHeader com.mbientlab.metawear.impl.Response.header' on a null object reference

### ISSUE THREAD

**Issue by gnanda17** [mbientlab/Metawear-AndroidAPI#9]
I need to set  mbient meta detector trackers for regular logging of telemetry. I see the below error consistently on every connect. The first time that I see it is when I am configuring the sensors for logging. I have made a simpler app with just my configuration code that consistently causes the below error. I would appreciate if you could let me know what might be causing this exception. 

On a few occasions I have seen an app crash (could be unrelated) but wanted to see if  you help better understand what is the cause of the below error and how one could avoid it. 

Link to source code:
https://github.com/gnanda17/configuring-metawear-board/blob/master/app/src/main/java/com/test/configuretrackers/ConfigurationApplication.java

Link to logcat gist:
https://gist.github.com/gnanda17/bbd6873b9d8ca71e32f1344ea4d30660

Configuring trackers:
Meta Detector with Temp + Proximity + Light + Acc

Error
```java
12-30 21:53:31.545 14234-14263/com.test.configuretrackers E/MetaWear: Background task reported an error
                                                                      java.util.concurrent.ExecutionException: java.lang.NullPointerException: Attempt to read from field 'com.mbientlab.metawear.impl.ResponseHeader com.mbientlab.metawear.impl.Response.header' on a null object reference
                                                                          at java.util.concurrent.FutureTask.report(FutureTask.java:94)
                                                                          at java.util.concurrent.FutureTask.get(FutureTask.java:164)
                                                                          at com.mbientlab.metawear.MetaWearBleService$3.run(MetaWearBleService.java:1364)
                                                                          at java.util.Timer$TimerImpl.run(Timer.java:284)
                                                                       Caused by: java.lang.NullPointerException: Attempt to read from field 'com.mbientlab.metawear.impl.ResponseHeader com.mbientlab.metawear.impl.Response.header' on a null object reference
                                                                          at com.mbientlab.metawear.impl.DefaultMetaWearBoard.receivedResponse(DefaultMetaWearBoard.java:4604)
                                                                          at com.mbientlab.metawear.MetaWearBleService$2$7.run(MetaWearBleService.java:1141)
                                                                          at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:423)
                                                                          at java.util.concurrent.FutureTask.run(FutureTask.java:237)
                                                                          at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1113)
                                                                          at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:588)
                                                                          at java.lang.Thread.run(Thread.java:818)
```

**Comment by scaryghost** [comment]
What API version are you using?

**Comment by gnanda17** [comment]
2.8.5

**Comment by mbientlab** [comment]
See if the latest changes on master branch fix the null pointer exception for your code

**Comment by scaryghost** [comment]
See if the latest changes on the master branch fix the null pointer exception for you

**Comment by gnanda17** [comment]
Hey thanks just tested my app out with the master branch. Its fixed. I no longer see the null pointer exception.

**Comment by scaryghost** [comment]
Good to hear.  The fix has been published as release v2.8.6.
